docs/modules/candid-guide/index.adoc

```diff
@@ -289,23 +320,56 @@ If you want to create a new Candid implementation, this document will not suffic
 
 Additionally, the official https://github.com/dfinity/candid/tree/master/test[Candid test data] is useful to make sure your implementation is compatible with Candid, even in slightly more obscure corner cases.
 
-[#reference]
 == Reference
 
-This section lists all the types supported by Candid, including their type syntax, the syntax for the textual representation, their upgrading rules and the corresponding types in Rust, Motoko and Javascript.
+This section provides links to Candid tools and libraries for currently supported host languages and detailed reference information about Candid supported types.
+
+If you are looking for more information before you start working with Candid, check out the following related resources:
+
+* link:https://www.youtube.com/watch?v=O2KaWRtsqHg[How Candid provides a common language for application interfaces (video)]
+
+=== Tools and libraries
+
+This section provides links to Candid tools and libraries for currently-supported host languages.
+
+* The Motoko language has built-in support for Candid and the compiler uses it implicitly.
+
+* To create Candid services or clients in Rust, you can use the https://crates.io/crates/candid[Candid crate].
+
+* The https://github.com/dfinity/agent-js/[Internet Computer agent library for JavaScript] includes support for Candid.
+
+* Community-supported libraries for Candid also exist for https://hackage.haskell.org/package/candid[Haskell], https://github.com/chenyan2002/ic-elm/[Elm] and https://github.com/seniorjoinu/candid-kt[Kotlin].
+
+* The https://github.com/dfinity/candid/tree/master/tools/ui[Candid UI] creates a Web interface from the Candid interface of any canister.
+
+=== Supported types
+
+This section lists all the types supported by Candid.
+For each type, the reference includes the following information:
+
+* Type syntax and the syntax for the textual representation of the type.
+* Upgrade rules for the type.
+* Corresponding types in Rust, Motoko and Javascript.
+
+== Subtypes and supertypes
```

Comment:
Starting a new section here is a bit of an odd break (and it should probably be a subsection).
```suggestion
Upgrade rules are given in terms of the possible _subtypes_ and _supertypes_ of a type.
```